components/AddressInput.js

```diff
@@ -0,0 +1,143 @@
+import { useState, useEffect, useRef } from 'react'
+import Select from 'react-select'
+import { useTranslation } from 'next-i18next'
+import { useRouter } from 'next/router'
+
+import {
+  isValidCTID,
+} from '../utils'
+
+import { userOrServiceLink } from '../utils/format'
+
+let typingTimer
+
+export default function AddressInput({ searchPlaceholderText, userData = {}, setFilters, type, inputValue, title, link }) {
```

Comment:
userData - this is usually an authorized user's address. 

addressInput is intended to be used for fields like 
receiver, issuer, and seat—so most likely it won't be the same user as the one that is logged in. 
Therefore, probably won't need here userData.